src/ImageEmbedPlugin.php

```diff
@@ -71,22 +71,66 @@ protected function embedImages(Swift_Mime_SimpleMessage $message, Swift_Mime_Sim
              * see https://github.com/swiftmailer/swiftmailer/issues/139
              */
             if (strpos($src, 'cid:') === false) {
+                $path = $this->getPathFromSrc($src);
 
-                $path = $src;
+                if ($this->fileExists($path)) {
+                    $entity = \Swift_Image::fromPath($path);
+                    $message->setChildren(
+                        array_merge($message->getChildren(), [$entity])
+                    );
 
-                if (filter_var($src, FILTER_VALIDATE_URL, FILTER_FLAG_SCHEME_REQUIRED) === false) {
-                    $path = $this->basePath . $src;
+                    $image->setAttribute('src', 'cid:' . $entity->getId());
                 }
+            }
+        }
 
-                $entity = \Swift_Image::fromPath($path);
-                $message->setChildren(
-                    array_merge($message->getChildren(), [$entity])
-                );
+        return $dom->saveHTML();
+    }
+
+    /**
+     * @param $path
+     *
+     * @return bool
+     */
+    protected function isUrl($path)
+    {
+        return filter_var($path, FILTER_VALIDATE_URL, FILTER_FLAG_SCHEME_REQUIRED) !== false;
+    }
 
-                $image->setAttribute('src', 'cid:' . $entity->getId());
+    /**
+     * @param $src
+     *
+     * @return string
+     */
+    protected function getPathFromSrc($src)
+    {
+        if ($this->isUrl($src)) {
+            return $src;
+        }
+
+        return $this->basePath . $src;
+    }
+
+    /**
+     * @param $path
+     *
+     * @return bool
+     */
+    protected function fileExists($path)
+    {
+        if ($this->isUrl($path)) {
+            $file_headers = @get_headers($path);
+
+            if($file_headers[0] != 'HTTP/1.0 200 OK'
+                || ($file_headers[0] == 'HTTP/1.0 302 Found'
+                    && $file_headers[7] != 'HTTP/1.0 200 OK')){
```

Comment:
Are you certain that this will always be the index 7? 

https://secure.php.net/manual/en/function.get-headers.php doesn't assure this.
I suppose that every server could insert extra headers before.

Can't you look at the beginning of each header instead of putting the index in the code?